### PR TITLE
Okky tp2

### PR DESCRIPTION
**Pull request by okkycorp**
## What changes are you trying to make? (e.g. Adding or removing code, refactoring existing code, adding reports)
Adding CC fraud detection using SMOTE to oversample minority and XGBoost. 
Achieved high accuracy and f1 score.
Added visualizations.

## What did you learn from the changes you have made?
SMOTE works great to create synthetic fraud data and as a result the XGBoost model were able to achieve high accuracy and f1 score.

## Was there another approach you were thinking about making? If so, what approach(es) were you thinking of?
I was going to do hyperparameter tuning using 5 fold cross validation grid search but was too resource intensive and does not have enough compute power to complete the task using XGBoost.

## Were there any challenges? If so, what issue(s) did you face? How did you overcome it?
Discovering the mechanism behind XGBoost and how it works, as well as researching the syntax and codes to use for the model to run well, and tweaking the random state to achieve the highest accuracy and f1 score.

## Checklist
- [ x] I can confirm that my changes are working as intended

### CODE REVIEW

**Comment by GibranAlvarez** [comment]
Great job! 👍🏾
